### PR TITLE
Feature presidecms 1311 datamanager select all

### PR DESCRIPTION
**Pull request by annbelievable**
This is to make sure the select all is there after any changes to the table body.

### CODE REVIEW

File: system/assets/js/admin/specific/datamanager/object/batchUpdateButton.js [hunk 1]
```diff
@@ -21,4 +21,25 @@
...
 	}
 
+	// select the target node
+	var target = document.querySelector('table');
+
+	// create an observer instance
+	var observer = new MutationObserver(function(mutations) {
+		mutations.forEach(function(mutation) {
+			if ( mutation.type === 'childList' ) {
+				console.log( "mutate" );
```
Comment (DominicWatson) [inline system/assets/js/admin/specific/datamanager/object/batchUpdateButton.js]: Console.log :)

File: system/assets/js/admin/specific/datamanager/object/batchUpdateButton.js [hunk 2]
```diff
@@ -21,4 +21,25 @@
 		} );
 	}
 
+	// select the target node
+	var target = document.querySelector('table');
```
Comment (DominicWatson) [inline system/assets/js/admin/specific/datamanager/object/batchUpdateButton.js]: Is there any reason this is not `$( ".object-listing-table" ).get( 0 )`? Seems odd to watch _all_ tables?

File: system/assets/js/admin/specific/datamanager/object/batchUpdateButton.js [hunk 3]
```diff
@@ -21,4 +21,25 @@
...
+
+	// create an observer instance
+	var observer = new MutationObserver(function(mutations) {
+		mutations.forEach(function(mutation) {
+			if ( mutation.type === 'childList' ) {
+				console.log( "mutate" );
+				var $listingTable  = $( '.object-listing-table' );
+				var $selectAllCBox = $listingTable.find( "th input:checkbox" );
+				$selectAllCBox.click();
```
Comment (DominicWatson) [inline system/assets/js/admin/specific/datamanager/object/batchUpdateButton.js]: Do we really want to click it any time there is a change in the table? If it is _unchecked_, does it then become selected? I think we need logic to only uncheck it when it is already checked.

**Review by DominicWatson — CHANGES_REQUESTED** [review]
Hi @annbelievable , thanks for the PR :). I've left a few review comments, could you take a look?

**Comment by annbelievable** [comment]
Hi @DominicWatson ,

I have made some correction, please kindly review again.

Thank you,
Kok Ann

**Comment by DominicWatson** [comment]
Great stuff, working perfectly. Thanks @annbelievable. Will be released with 10.9.6.

**Comment by annbelievable** [comment]
Hi @DominicWatson ,

Please do include in the 10.9 version too as it affects the older version

**Pull request merged**